### PR TITLE
4 new spanish providers and extractors

### LINKED ISSUE #344: (Spanish) https://cuevana3.io/
https://cuevana3.io/ has several servers with Spanish dub and subtitles, its Fembed server is fast. Is it posible? Thnx

### PR DESCRIPTION
**Pull request by Stormunblessed**
Cuevana3.io, Pelisflix.li, Seriesflix.video, EntrePeliculasySeries.nu, Ok.ru extractor, and tomatomatela is required for cuevana.
Closes #344

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/extractors/OkRuExtractor.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.lagradost.cloudstream3.extractors
+
+import com.lagradost.cloudstream3.utils.*
+import com.lagradost.cloudstream3.app
+
+class Okru1 : OkRu() {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/extractors/OkRuExtractor.kt:6]: Okru1 has the same url as Okru

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/PelisflixProvider.kt [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            }
+            if (list.isEmpty()) throw ErrorLoadingException("No Seasons Found")
+
+            val episodeList = ArrayList<TvSeriesEpisode>()
+
+            for (season in list) {
+                val seasonDocument = app.get(season.second).document
+                val episodes = seasonDocument.select("table > tbody > tr")
+                if (episodes.isNotEmpty()) {
+                    episodes.apmap { episode ->
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/PelisflixProvider.kt:129]: as you dont do any get request here, you should use .forEach

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/SeriesflixProvider.kt [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                        TvType.Movie,
+                        it.selectFirst("figure img").attr("src"),
+                        null,
+                        null,
+                    )
+                }
+
+                items.add(HomePageList(i.second, home))
+            } catch (e: Exception) {
+                e.printStackTrace()
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/SeriesflixProvider.kt:44]: logError(e)

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/SeriesflixProvider.kt [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+                        "Sec-Fetch-Dest" to "iframe",
+                        "Sec-Fetch-Mode" to "navigate",
+                        "Sec-Fetch-Site" to "same-origin",
+                        "Sec-Fetch-User" to "?1",),
+                    params = mapOf(Pair("h", postkey)),
+                    data =  mapOf(Pair("h", postkey)),
+                    allowRedirects = false
+                ).response.headers.values("location").apmap {link ->
+                    val url1 = link.replace("#bu","")
+                    for (extractor in extractorApis) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/SeriesflixProvider.kt:215]: loadExtractor

**Comment by Stormunblessed** [comment]
I think that's it

**Review by LagradOst — COMMENTED** [review]
Quite a big pr, I might have missed smth. (I accidentally clicked add comment instead of start review)

But general notes, only really use apmap for parallel network requests and .forEach or .map for everything else.

use loadExtractor instead of doing the same thing yourself

Inline variable getter so you dont have get() =

Try to avoid regex if you can and instead use css selectors so other people can update the provider easier.

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/PelisflixProvider.kt [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            .replace(descRegex4, "").replace(descRegex5, "")
+        val desc2Regex = Regex("(G(e|é)nero:.*..)")
+        val descipt2 = document.selectFirst("div.Description").text().replace(desc2Regex,"")
+        val rating =
+            document.selectFirst("div.rating-content button.like-mov span.vot_cl")?.text()?.toFloatOrNull()
+                ?.times(0)?.toInt()
+        val year = document.selectFirst("span.Date")?.text()
+        val duration = if (type == TvType.Movie) document.selectFirst(".Container .Container  span.Time").text() else null
+        val postercss = document.selectFirst("head").toString()
+        val posterRegex = Regex("(\"og:image\" content=\"https:\\/\\/seriesflix.video\\/wp-content\\/uploads\\/(\\d+)\\/(\\d+)\\/?.*.jpg)")
```
[the remark this answers is not in the document]
Comment (Stormunblessed) [reply app/src/main/java/com/lagradost/cloudstream3/movieproviders/PelisflixProvider.kt:103]: Tried to, but gives me a blank image

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/SeriesflixProvider.kt [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+                val href = element.attr("href")
+                if (season != null && season > 0 && !href.isNullOrBlank()) {
+                    list.add(Pair(season, fixUrl(href)))
+                }
+            }
+            if (list.isEmpty()) throw ErrorLoadingException("No Seasons Found")
+
+            val episodeList = ArrayList<TvSeriesEpisode>()
+
+            for (season in list) {
```
[the remark this answers is not in the document]
Comment (Stormunblessed) [reply app/src/main/java/com/lagradost/cloudstream3/movieproviders/SeriesflixProvider.kt:126]: Don't know how to write this one lmao

**Pull request merged**